static/js/containers/DashboardPage.js

```diff
@@ -119,14 +111,16 @@ class DashboardPage extends React.Component {
     dispatch:                 Dispatch,
     setCalculatorVisibility:  (b: boolean) => void,
     ui:                       UIState,
-    courseTeamEmail:          EmailState,
+    email:                    AllEmailsState,
     documents:                DocumentsState,
     fetchDashboard:           () => void,
     orderReceipt:             OrderReceiptState,
     courseEnrollments:        CourseEnrollmentsState,
     financialAid:             FinancialAidState,
     location:                 Object,
     pearson:                  PearsonAPIState,
+    emailDialog:              React$Element<*>|null,
+    emailOpenActions:         {[emailType: string]: () => void}
```

Comment:
it also strikes me as a little cleaner to have the HOC only pass down the action which is relevant to the particular wrapped component (indicated by the `emailType` argument passed to the HOC), so that the wrapped component doesn't have to worry about the logic. Although maybe in the future we would want to support opening multiple different email types, but then again if we wanted to add that behavior later it would just be a small refactor.